### PR TITLE
Add --iree-codegen-linalg-max-constant-fold-elements= flag.

### PR DESCRIPTION
**Pull request by benvanik**
Currently set to intmax so there's no changes to users models. Users can opt in to disabling the extremely slow transpose folder as though it's critical for reducing runtime memory consumption it is really bad for iteration time on large models. Having a way to bound the folding is useful in lieu of a real solution (fast transpose in C++, consteval, whatever).

Workaround until #12033 (or another) does something better.

### CODE REVIEW

**Review by MaheshRavishankar — APPROVED** [review]
Ooh! Nice! Can't believe I didn't think of this!

**Comment by benvanik** [comment]
Ditto :P

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp [hunk 1]
```diff
@@ -291,6 +298,13 @@ struct FusionOfTensorOpsPass
 
       // Constant fold Linalg operations.
       auto constantFoldControlFn = [](OpOperand *fusedOperand) {
+        if (auto shapedType =
+                fusedOperand->get().getType().dyn_cast<ShapedType>()) {
```
Comment (jpienaar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: Nit: free standing cast is the form now.

**Review by jpienaar — APPROVED** [review]
Looks good as escape valve, thanks!

**Pull request merged**